src/EnergyPlus/HVACManager.cc

```diff
@@ -2725,6 +2730,46 @@ namespace HVACManager {
 
 	}
 
+	void
+	CheckMassBalances()
+	{
+		using namespace Fans;
+		using namespace Pumps;
+		using General::RoundSigDigits;
+		using DataHVACGlobals::SmallMassFlow;  // Air
+		using DataBranchAirLoopPlant::MassFlowTolerance;  // Liquid
+
+		for (int FanNum = 1; FanNum <= NumFans; ++FanNum) {
+
+//			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Fan( FanNum ).InletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan inlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).InletAirMassFlowRate, 4) + ") does not match inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+//
+//			if (abs(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate - Fan( FanNum ).OutletAirMassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Fan( FanNum ).OutletAirMassFlowRate, 4) + ") does not match outlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ").");
+//			}
+
+			if (abs(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate - Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate) > SmallMassFlow) {
+//				ShowFatalError(Fan(FanNum).FanType + ", Name: " + Fan(FanNum).FanName + ": Fan outlet mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).OutletNodeNum ).MassFlowRate, 4) + ") does not match fan inlet node mass flow rate (" + RoundSigDigits(Node( Fan( FanNum ).InletNodeNum ).MassFlowRate, 4) + ").");
+// can we get a handle on why this is happening before we start crashing user files?
```

Comment:
@nealkruis Lots of components grab various inlet and outlet data into the component data structure to use for calculations and reporting.  In a really small nutshell, the concept was initialize the component model with inlet conditions (and control info), do the calculations, and then update the outlet node data.  I think that's still appropriate.  But some components just use local variables to hold this during the calculations, others keep it in the structure.  Sometimes it's there to support component-level output variables.  Sometimes it's in the component structure just because that was the developers style.  Clearly a topic to standardize on as we refactor things.  in this case, It could be helpful to identify if/when something else is touching the node mass flow after the fan is done.